File: Haskell/problem004.hs
{-
Problem 4 - Project Euler
http://projecteuler.net/index.php?section=problems&id=4
-}

isPalindromic :: Int -> Bool
isPalindromic x =
     let s = show x
     in  s == reverse s

problem004 = maximum [(x * y, x, y) | x <- [100..999], y <- [100..x], isPalindromic (x * y)]

main = putStrLn $ "problem004: answer " ++ show problem004

File: Go/problem004.go
package projecteuler

import (
	"log"
	"strconv"
)

// Problem 4 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=4

func isPalindromic(x int) bool {
	s := strconv.Itoa(x)
	length := len(s)
	for i := 0; i < length/2; i++ {
		if s[i] != s[length-i-1] {
			return false
		}
	}
	return true
}

func P004() {
	ans := 0
	for i := 100; i < 1000; i++ {
		for j := i; j < 1000; j++ {
			x := i * j
			if ans < x && isPalindromic(x) {
				ans = x
			}
		}
	}
	log.Println(ans)

}
